openmdao/core/component.py

```diff
@@ -1099,8 +1099,8 @@ def declare_partials(self, of, wrt, dependent=True, rows=None, cols=None, val=No
         if dependent:
             meta['value'] = val
             if rows is not None:
-                rows = np.array(rows, dtype=INT_DTYPE, copy=False)
-                cols = np.array(cols, dtype=INT_DTYPE, copy=False)
+                rows = np.array(rows, dtype=INT_DTYPE, copy=False).flatten()
+                cols = np.array(cols, dtype=INT_DTYPE, copy=False).flatten()
```

Comment:
Ok I have a fix in for this. I'm flattening inside of `Demux`'s `add_var`. T he current code only supported 2D and below to have the same length of rows and cols because `atleast_1d` preserves the dimensions and promotes scalers to 1D matrices. Since we're passing in 3D or higher arrays, the final answer needs to be flattened but `atleast_1d` needs to stay for the 1D case because it results in a `TypeError: len() of unsized object` if not